lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -692,6 +694,103 @@ class DecomposeAtenStdOp : public OpRewritePattern<AtenStdOp> {
 };
 } // namespace
 
+// Determine uniform lower and upper bounds given probability p.
+// The lower and upper bounds lie in range [0, 1].
+static std::pair<double, double> giveUniformBounds(double p) {
+  std::pair<double, double> bounds{0.0, 1.0};
+  if (p < 0.5)
+    bounds.second = (p * 2.0) + bounds.first;
+  if (p > 0.5)
+    bounds.first = (p * 2.0) - bounds.second;
+  return bounds;
+}
+
+// Returns a tensor with bernoulli(p) distribution.
+// Decompose aten.bernoulli(x, p) to aten.gtTensor(aten.uniform(x), x).
+static Value decomposeBernoulliLikeOp(PatternRewriter &rewriter, Operation *op,
+                                      Location loc, Value input, double p) {
+  BaseTensorType inputType = input.getType().cast<BaseTensorType>();
+
+  // `intType` contains the corresponding integer for the dtype which is used
+  // by the aten.to.dtype op.
+  int intType = (int)getScalarTypeForType(inputType.getDtype());
+  Value convertIntVal =
+      rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(intType));
+  if (!inputType.hasSizes())
+    return nullptr;
+  BaseTensorType boolType =
+      inputType
+          .getWithSizesAndDtype(
+              inputType.getSizes(),
+              IntegerType::get(op->getContext(), 1, IntegerType::Signless))
+          .cast<BaseTensorType>();
+  // Get the upper and lower bounds for the given probability.
+  auto uniformBounds = giveUniformBounds(float(p));
+  Value lb = rewriter.create<ConstantFloatOp>(
+      loc, rewriter.getF64FloatAttr(uniformBounds.first));
+  Value ub = rewriter.create<ConstantFloatOp>(
+      loc, rewriter.getF64FloatAttr(uniformBounds.second));
+
+  Value falseVal = rewriter.create<ConstantBoolOp>(loc, false);
+  Value noneVal = rewriter.create<ConstantNoneOp>(loc);
+
+  // Create a uniform random op with low and high set to lb and ub respectively.
+  Value uniformRandom = rewriter.create<PseudoAtenUniformOp>(
+      loc, inputType, input, lb, ub, noneVal);
+  Value gtValue =
+      rewriter.create<AtenGtTensorOp>(loc, boolType, input, uniformRandom);
```

Comment:
I just realize that the uniformRandom result should be compared against `p` rather than input. `Fills each location of self with an independent sample from `Bernoulli(p)`. `Bernoulli` is `takes the value 1 with probability p and the value 0 with probability 1-p`. The comparison result should be AtenLTScalar(uniformRandom, p).